### DISCUSSION TITLE
Question about IOServiceProvider and files that evolve over time

### DISCUSSION THREAD

**Discussion by francoisandre** [Unidata/netcdf-java#1134] [Questions and Answers]
Hello,

I'm really really sorry to bother you.
I am trying to set up an IOServiceProvider to extract time series from my data CSV-files via THREDDS WMS service and NCML. 

It works fine when my data file is static (does not change over time). 

But I don't know what is the right way to handle files that change in real time and how to reload the data regularly. I thought that the syncExtend method was used to do this and that this would be the place to calculate the new size of the "time" dimension. But this method is deprecated and is never called by the upper layers. So the data loaded the first time stays in the cache and is never completed.

Is it up to my IOSP to do this reload itself regularly ? Should I set up the cache system differently ?

Thank you very much for any help because I have both the feeling to be close to the solution while thinking I have understood absolutely nothing :-)

Sincerely

François

PS: 
- To test, I am using CDM 5.5.4-SNAPSHOT and EDAL 1.5.0.6-SNAPSHOT (which seems to me to be the versions used by the latest version of THREDDS).
- My IOSP code is here: https://github.com/sedoo/cdm-csv-module

**Comment by haileyajohnson** [comment]
Hello, no problem for the question! To reload the data you'll need to reopen the `netcdfFile` or `netcdfDataset`. There's no explanation that I can see on why `syncExtend` was deprecated, but my guess would be because we're moving toward immutability and don't want to allow open `netcdfFile` objects to change.
